### PR TITLE
Support testing library/agent instrumentation interoperability

### PR DESCRIPTION
**Pull request by trask**
I don't think the removed code is needed anymore since the library instrumentation is shaded inside of the agent even during tests now. But let's see what CI says...

Removing this code will allow agent tests to be written that use the library instrumentation and verify interoperability between the two.

### CODE REVIEW

**Comment by anuraaga** [comment]
@trask Looks like CI might not like this PR, but also the tests not catching https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4718 makes me think that either an improvement or alternative to this snippet is still needed

**Comment by anuraaga** [comment]
@trask Thanks for looking at this :) By the way, the idea I had in mind to remove this snippet and make things hopefully more robust is a custom `configuration` type for instrumentation dependencies we bundle. We currently have `library`, which is like `compileOnly` + `testImplementation` (with the ability to raise the version number to latest), we could have a `bundled` which is like `compileOnly` + `shadowInclude` (semantically, we don't shade each individual project so this configuration would need to be recognized at the bundle site). And this would then make sure the files aren't in the test classpath in a more Gradle-native way than my hack

**Comment by trask** [comment]
> the idea I had in mind to remove this snippet and make things hopefully more robust is a custom `configuration` type for instrumentation dependencies we bundle. We currently have `library`, which is like `compileOnly` + `testImplementation` (with the ability to raise the version number to latest), we could have a `bundled` which is like `compileOnly` + `shadowInclude` (semantically, we don't shade each individual project so this configuration would need to be recognized at the bundle site). And this would then make sure the files aren't in the test classpath in a more Gradle-native way than my hack around our current usage of `implementation`

This sounds cool, though I'll probably leave this part for the gradle magicians ✨

I think I've about worked out what should / should not be on the class path in order to enable writing tests that verify interoperability between agent and library instrumentation (see also #4779).

**Comment by trask** [comment]
CI still hasn't passed yet, but I'm feeling good about this next run, so optimistically marking it ready for review 😂

File: conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts [hunk 1]
```diff
@@ -106,22 +106,24 @@ afterEvaluate {
...
       // plugin identification instead.
 
-      // If agent depends on some shared instrumentation module that is not a testing module, it will
-      // be packaged into the testing jar so we need to make sure to exclude from the test classpath.
       val lib = it.absoluteFile
-      val instrumentationDir = file("${rootDir}/instrumentation/").absoluteFile
-      if (lib.startsWith(instrumentationDir) &&
-        lib.extension == "jar" &&
-        !lib.absolutePath.substring(instrumentationDir.absolutePath.length).contains("testing")) {
+      if (lib.name.startsWith("opentelemetry-javaagent-")) {
```
Comment (iNikem) [inline conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts:119]: If `lib` is an `absoluteFile`, how can it start with `opentelemetry-javaagent`?

Comment (trask) [reply conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts:119]: it's `lib.name`, e.g. `File.getName()`, not the full path

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**